src/rpp/rpp/operators/on_error_resume_next.hpp

```diff
@@ -0,0 +1,109 @@
+
+        mutable TObserver observer;
+        Selector          selector;
+        // Manually control disposable to ensure observer is not used after move in on_error emission
+        mutable rpp::composite_disposable_wrapper disposable = composite_disposable_wrapper::make();
+
+        RPP_CALL_DURING_CONSTRUCTION(
+            {
+                observer.set_upstream(disposable);
+            });
+
+        template<typename T>
+        void on_next(T&& v) const
+        {
+            observer.on_next(std::forward<T>(v));
+        }
+
+        void on_error(const std::exception_ptr& err) const
+        {
+            disposable.dispose();
+            selector(err).subscribe(std::move(observer));
+        }
+
+        void on_completed() const
+        {
+            disposable.dispose();
+            observer.on_completed();
+        }
+
+        void set_upstream(const disposable_wrapper& d)
+        {
+            disposable.add(d);
+        }
+
+        bool is_disposed() const { return disposable.is_disposed(); }
+    };
+
+    template<rpp::constraint::decayed_type Selector>
+    struct on_error_resume_next_t : lift_operator<on_error_resume_next_t<Selector>, Selector>
+    {
+        template<rpp::constraint::decayed_type T>
+        struct operator_traits
+        {
+            using selector_observable_result_type =
+                std::decay_t<rpp::utils::extract_observable_type_t<std::invoke_result_t<Selector, std::exception_ptr>>>;
+
+            static_assert(
+                std::is_same_v<selector_observable_result_type, T>,
+                "Selector observable result type is not the same as T");
+
+            using result_type = T;
+
+            template<rpp::constraint::observer_of_type<result_type> TObserver>
+            using observer_strategy = on_error_resume_next_observer_strategy<TObserver, Selector>;
+        };
+
+        template<rpp::details::observables::constraint::disposable_strategy Prev>
+        using updated_disposable_strategy = rpp::details::observables::fixed_disposable_strategy_selector<1>;
+    };
+} // namespace rpp::operators::details
+
+namespace rpp::operators
+{
+    /**
+     * @brief If an error occurs, take the result from the Selector and subscribe to that instead.
+     *
+     * @marble on_error_resume_next
+       {
+           source observable                                  : +-1-x
+           operator "on_error_resume_next: () => obs(-2-3-|)" : +-1-2-3-|
+       }
+     *
+     * @param selector callable taking a std::exception_ptr and returning observable to continue on
+     *
+     * @warning #include <rpp/operators/on_error_resume_next.hpp>
+     *
+     * @ingroup error_handling_operators
+     * @see https://reactivex.io/documentation/operators/catch.html
+     */
+    template<typename Selector>
+        requires (!utils::is_not_template_callable<Selector> || rpp::constraint::observable<std::invoke_result_t<Selector, std::exception_ptr>>)
```

Comment:
in this case we don't need chaeck over `is_not_template` because we know that it should be callable with exception_ptr. This check needed when we don't know final type (for example, map)

Reply:
done